backends/cuda/runtime/cuda_backend.cpp

```diff
@@ -0,0 +1,374 @@
+/*
```

Comment:
Why not keep bulk of code under `backend/aoti` and keep only cuda specific runtime AOTI bits here? Rationale is code dedup across all aoti backends.